### ISSUE TITLE
Map Orientation

### ISSUE THREAD

**Issue by sek19** [mapbox/mapbox-unity-sdk#1348]
* Support
* Unity version: 2018.3.9f1
* Mapbox SDK version: 2.0.0
* The platform you're building to: Android and IOS
* A description of what you're trying to do: 

What I am trying to do is, align the world scaled ArAlignedMap to the orientation and rotation of a known location from an image target.

The way I've been approaching this is by setting the map location to a known long and lat cord AKA my image target. I'm centering the map to the image target so that I can rotate the world `_map` around it's known location.

`_map.SetCenterLatitudeLongitude(markerLongLat);
        _map.SetPlacementType(MapPlacementType.AtLocationCenter);`

Where I have my issue is when orienting the rotation of the `_map.Root.Rotation` . My current implementation uses the `ARcamera.rotation.y` value to rotate the world with the camera, I offset this manually based on each marker location since I know that the north position of that marker is not changing. Since the map is basing it's `_map.rotation.y` and offset rotation from the camera I get close to true alignment.  But if the camera is not centered exactly parallel to the image target forward position; the alignment is off. I feel like this is close to the implementation I should be using but wanted to see if you have a better suggestion on how I would approach adjusting the rotation of the map. The rotation code is as follows:

`_map.Root.rotation = Quaternion.Euler(0, arCamera.transform.rotation.eulerAngles.y - offsetRotation.eulerAngles.y , 0);`

Some things that I've been looking for in the Documentation and minor questions that you may know better than I.

what would I use to find the north position on a map that has been generated?

since I have a known location and it's true world north rotation; is there a function I can call to align the map based on it's long, lat , rotation based on a north position (0-360)

I hope this explains my problem, thanks for taking the time to read though this as I know it's a bit lengthy.

**Comment by brnkhy** [comment]
@sek19 admittely I'm a little confused but might have a few ideas; when I need to rotate map I generally do not rotate the map object itself (as it might cause some confusion), I move map under another empty gameobject and then rotate that new root. So original map values and my new additional rotation is applied separately, keeping it all cleaner and easier to read/debug.
This would also help with finding the 'north' direction. Z+ is north by default but of course it might get little tricky when you rotate the map. If you root a parent object instead of the map object itself though, map object local z+ will still be the north. Then you can convert that local vector to world space if you want as well.
Hope that helps!

**Comment by sek19** [comment]
@brnkhy Thank you for helping to clarify the true north position, it really helped me narrow down what was going wrong in the orientaiton.

So where I was going wrong in my approach was that my image targets forward position was actually in the y direction so I switched around the Quaternions to have the y axis become the z axis. 

This let me use the image target as a basis for an offset. 

so the original code:

`_map.Root.rotation = Quaternion.Euler(0, arCamera.transform.rotation.eulerAngles.y - offsetRotation.eulerAngles.y , 0);`

then became 

`_map.Root.rotation = Quaternion.AngleAxis(arCamera.transform.eulerAngles.y, Vector3.up) * Quaternion.AngleAxis(-100f, Vector3.up) * offsetRotation;`

Since I was getting consistent results by using the AR camera as the map rotation all I had to do was take the ARCamera.rotation and multiply that by a constant of -100 because the real world orientation of the image target was offset north by that amount. Then multiplied an offsetRotation (offset rotation being my offset from the image target and the ARCamera) so for example. if I was standing right of the image target the offset became ARCameraRot * -° and the inverse was true by standing on the opposite side. This narrowed me into a very consistent Rotation of the map based on the position and rotation of the image target.
